### PR TITLE
CATL-1310: Replace home menu

### PR DESCRIPTION
**Pull request by reneolivo**
## Overview
This PR replaces the CiviCRM's Home menu with a custom one that has the following features:

- It removes the sub menu.
- It adds a label as defined from the Navigation configuration.
- When clicked it redirects to the URL defined from the navigation configuration.

## Before
gif

## After
gif

## Technical Details

* We add a new Home Menu hook in `CRM/Usermenu/Hooks/CoreResourceList/HomeMenu.php `. This hook includes the `usermenu.js` file and also the information about the home navigation item, including the label and URL to use for replacing the menu. These are only included so long as the navigation item is enabled.
* `scss/components/outside-namespace/_crm-logo-label.scss` Adds a small margin left of `6px` to the home label. There are 2 importing notes here:
  1 - we add the margin to the label, not to the icon, so if the label is not added (by disabling the navigation item) we don't have an extra margin that is not needed.
  2 - the `6px` constant is used because there's no variable defined for it: https://github.com/civicrm/org.civicrm.shoreditch/blob/master/scss/civicrm/common/_main-menu.scss#L4

### CODE REVIEW

File: CRM/Usermenu/Hooks/CoreResourceList/HomeMenu.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
+<?php
```
Comment (deb1990) [inline CRM/Usermenu/Hooks/CoreResourceList/HomeMenu.php:1]: Please add the phplinter to this repo and fix the linting errors.

Comment (deb1990) [reply CRM/Usermenu/Hooks/CoreResourceList/HomeMenu.php:1]: Also add the JS and scss linters too.

File: js/homemenu.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+   * - Removes the dropdown menu.
+   * - Adds a custom label next to the CiviCRM icon.
+   * - Adds a link to the URL configured for the home menu item.
+   */
+  function replaceHomeMenu () {
+    var $homeMenuLinkClone = $homeMenuLink.clone();
+    $homeMenuLink.remove();
+
+    $homeMenuLinkClone
+      .append('<span class="crm-logo-label">' + homeMenuItem.label + '</span>')
```
Comment (deb1990) [inline js/homemenu.js]: Can we move the markup to a separate variable?

**Review by deb1990 — CHANGES_REQUESTED** [review]

File: CRM/Usermenu/Hooks/CoreResourceList/HomeMenu.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+   * Returns the Home Menu navigation item. Stores it in a local variable to avoid
+   * repeated calls to the database. Returns NULL when the navigation item is not
+   * found.
+   *
+   * @return Array|Null
+   */
+  private function getHomeMenu() {
+    if (!$this->homeMenu) {
+      $response = civicrm_api3('Navigation', 'get', [
+        'sequential' => 1,
```
Comment (tunbola) [inline CRM/Usermenu/Hooks/CoreResourceList/HomeMenu.php]: No need for sequential here.

File: CRM/Usermenu/Hooks/CoreResourceList/HomeMenu.php [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+   * Home Menu navigation item is set to active.
+   *
+   * @param Array $items
+   * @param String $region
+   * @return Bool
+   */
+  public function shouldRun(array &$items, $region) {
+    $homeMenu = $this->getHomeMenu();
+    $isHeaderRegion = $region === 'html-header';
+    $isUserMenuActive = $homeMenu && $homeMenu['is_active'] == 1;
```
Comment (tunbola) [inline CRM/Usermenu/Hooks/CoreResourceList/HomeMenu.php]: Should this be called `isHomeMenuActive` instead ?

**Review by tunbola — CHANGES_REQUESTED** [review]

File: CRM/Usermenu/Hooks/CoreResourceList/HomeMenu.php (hunk 1 above)
Comment (reneolivo) [reply CRM/Usermenu/Hooks/CoreResourceList/HomeMenu.php:1]: @tunbola the linters have been added in the previous PR. We rebased this one.

**Review by tunbola — APPROVED** [review]

**Pull request merged**